### ISSUE TITLE
After selecting all the correct images the captcha shows you, the recaptcha doesn't get re-rendered with a pass (green check) 

### ISSUE THREAD

**Issue by vascojm82** [dozoisch/react-google-recaptcha#132]
I'm adding a `<ReCaptcha />` component inside a redux-forms `<Field />` component, the below `<Field />` component is returned by the render function of the myForm component I have: 

`<Field name='captcha' size="normal" theme="dark" component={(formProps) => this.renderReCaptcha(formProps)} />`

This is my renderReCaptcha function that returns the `<ReCaptcha />` component:

```
renderReCaptcha = (formProps) => {
    return(
      <div>
        <ReCaptcha
            ref={(ref) => {this.captcha = ref;}}
            size={formProps.size}
            theme={formProps.theme}
            sitekey="<SOME-KEY>"
            onChange={this.onChange}
        />
      </div>
    );
  }
```

After selecting all the correct images the captcha shows you and clicking 'Verify', I get a recaptchaToken back which means it passed, however, the recaptcha doesn't get re-rendered with a pass (green check) like below:

![image](https://user-images.githubusercontent.com/18554186/53449645-4ab3bd00-39e8-11e9-8e1f-df9a930103e0.png)

instead it goes back to:

![image](https://user-images.githubusercontent.com/18554186/53449691-6ae37c00-39e8-11e9-9233-d705b1a07283.png)

Is that what's supposed to happen?.  On the [codesandbox.io](https://codesandbox.io/s/1y4zzjq37l) example as soon as you click on it it goes to:

![image](https://user-images.githubusercontent.com/18554186/53449822-cc0b4f80-39e8-11e9-8f10-5242efb71928.png)

**Comment by dozoisch** [comment]
It seems like the component gets unmounted and remounted

**Comment by vascojm82** [comment]
When I finish choosing the images and click 'verify' only the recaptcha flickers and it stays normal like:

![image](https://user-images.githubusercontent.com/18554186/53451085-f3175080-39eb-11e9-9153-60baa9c51758.png)

**Comment by vascojm82** [comment]
Could it be that something changing in the state is causing React to re-render something at the same time and that's affecting the reCaptcha?.  What could be triggering the component to unmount and then remount?

**Comment by dozoisch** [comment]
Using react lifecycle functions, try to pinpoint if components near the captcha gets re-mounted. One thing to do could be to wrap the reCaptcha in a class React.Component to be able to get access to lifecycle function.

**Comment by vascojm82** [comment]
It looks like every time I type something in the form, it re-renders the component and when that happens the reCaptcha goes from:

![image](https://user-images.githubusercontent.com/18554186/53453256-96b72f80-39f1-11e9-9268-f8599efed189.png)

to just this again:

![image](https://user-images.githubusercontent.com/18554186/53453262-9b7be380-39f1-11e9-8a32-2c9491508f4f.png)

How can I make sure the reCaptcha doesn't get re-rendered when redux-form or react-redux-localize triggers a re-render.  I'm pretty sure that's the issue now.  I tried wrapping the ReCaptcha in a component before and it was doing the same, that's why I put it back in my form component.

**Comment by dozoisch** [comment]
Using Reatc.PureComponent and/or the lifecycle shouldComponentUpdate is the way to do it.

https://reactjs.org/docs/react-api.html#reactpurecomponent

**Comment by vascojm82** [comment]
I found the problem, yes the whole component was re-rendering because I was doing something like this:

`<CommentItem likeComment={() => this.likeComment(user.id)} />`

The problem is that every time the parent’s render method is called, a new function (with a new reference) is created to be passed to **likeComment**. This has the side effect of changing the props on each child which in turn will cause them all to re-render, even if the data itself is all the same.

To get around this, only pass the reference to the parent’s prototype method to children. The child’s **likeComment** prop will always have the same reference and never cause a needless re-render.

Changed it to something like this and it works fine now, no re-renders:

`<CommentItem likeComment={this.likeComment} userID={user.id} />`
